File: GrooveOn.Models/ResponseObjects/SubscriptionResponse.cs
using System;

namespace GrooveOn.Model.ResponseObjects
{
    public class SubscriptionResponse
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public string? Username { get; set; }

        public string? UserFullName { get; set; }

        public int SubscriptionPlanId { get; set; }

        public string? SubscriptionPlanName { get; set; }

        public float SubscriptionPlanPrice { get; set; }

        public int SubscriptionPlanDurationDays { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime? ExpiryDate { get; set; }

        public bool IsActive { get; set; }

        public bool IsExpired =>
            ExpiryDate.HasValue && ExpiryDate.Value < DateTime.UtcNow;
    }
}

File: UI/grooveon_mobile/lib/models/subscription_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'subscription_response.g.dart';

@JsonSerializable()
class SubscriptionResponse {
  final int id;
  final int userId;
  final String? username;
  final String? userFullName;
  final int subscriptionPlanId;
  final String? subscriptionPlanName;
  final double subscriptionPlanPrice;
  final int subscriptionPlanDurationDays;
  final DateTime startDate;
  final DateTime? expiryDate;
  final bool isActive;
  final bool isExpired;

  SubscriptionResponse({
    required this.id,
    required this.userId,
    this.username,
    this.userFullName,
    required this.subscriptionPlanId,
    this.subscriptionPlanName,
    required this.subscriptionPlanPrice,
    required this.subscriptionPlanDurationDays,
    required this.startDate,
    this.expiryDate,
    required this.isActive,
    required this.isExpired,
  });

  factory SubscriptionResponse.fromJson(Map<String, dynamic> json) =>
      _$SubscriptionResponseFromJson(json);

  Map<String, dynamic> toJson() => _$SubscriptionResponseToJson(this);
}